### PR TITLE
Keep base version etag during reload

### LINKED ISSUE #5726: Close has already been called on the connection
While I thought we had resolved this we still seem to try to close a connection that has already been closed regularly. I'm not sure if this is avoidable, but if not we can maybe handle that more gracefully instead of throwing and erroring out on the console

Screenshot 2024-04-24 at 16 33 45
Screenshot 2024-04-24 at 16 33 50

https://nextcloud-gmbh.sentry.io/share/issue/4eff0f8ac50b49d2a481d35e2f718ab0/

@max-nextcloud Any idea?

### PR DESCRIPTION
**Pull request by max-nextcloud**
### 📝 Summary

`this.$syncService` is cleared during the `close` method.

However we need the `baseVersionEtag` to ensure the editing session
on the server
is still in sync with our local ydoc.

See https://github.com/nextcloud/text/issues/5724#issuecomment-2180699808 for a trace of the effect of this.

Also fixes #5726 

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation is not required

### CODE REVIEW

**Comment by max-nextcloud** [comment]
I'm looking into the flaky sync tests and fixing some issues with lost network connectivity on the way.
In particular when clicking reconnect the editor and the status message go away now. If the reconnect fails there's nothing one can do anymore.
For now I'll focus on just keeping the status message so one can try to reconnect multiple times. Keeping the editor would be great but seems more involved.

File: src/services/SyncService.js [hunk 1]
```diff
@@ -270,6 +271,29 @@ class SyncService {
...
 
+	async sendRemainingSteps(queue) {
+		if (queue.length === 0) {
+			return
+		}
+		let outbox = []
+		const steps = queue.map(s => encodeArrayBuffer(s))
+			.filter(s => s < 'AQ')
+		const awareness = queue.map(s => encodeArrayBuffer(s))
+			.findLast(s => s > 'AQ') || ''
```
Comment (juliushaertl) [inline src/services/SyncService.js:282]: We we have this now in two places with services/WebSocketPolyfill.js maybe we put in a helper function? Could probably then also use a y.js method if available to filter (I'm not a fan of the magic strings there)...

But not a blocker

**Review by juliushaertl — APPROVED** [review]
Very nice, thanks also for the nicely split commits. Makes reviewing a very straight forward ❤️

**Review by mejo- — COMMENTED** [review]
Did some testing and seems to work well. Thanks so much for looking into it.

When cutting the network and pressing "Reconnect", we run into

`sendStepsNow(): this.connection is undefined` because the connection got closed but `sendRemainingSteps()` still tries to run `sendStepsNow()`.

In the same scenario `Error: Close has already been called on the connection` is also still logged.

**Comment by max-nextcloud** [comment]
> When cutting the network and pressing "Reconnect", we run into
> 
> `sendStepsNow(): this.connection is undefined` because the connection got closed but `sendRemainingSteps()` still tries to run `sendStepsNow()`.
> 
> In the same scenario `Error: Close has already been called on the connection` is also still logged.

I tried but failed to reproduce these two.

I also don't see how that would happen as reconnect should not send the remaining steps anymore. But I will take a look at the code to confirm.

File: src/components/Editor.vue [hunk 2]
```diff
@@ -432,7 +430,7 @@ export default {
 		reconnect() {
 			this.contentLoaded = false
 			this.hasConnectionIssue = false
-			this.close().then(this.initSession)
+			this.disconnect().then(this.initSession)
```
Comment (max-nextcloud) [inline src/components/Editor.vue:433]: `reconnect` calling `disconnect` rather than close now.

**Comment by max-nextcloud** [comment]
Merged as i could not reproduce the remaining issue. @mejo- Please open a new issue based on the comment if this still happens for you.

**Pull request merged**